src/dataExplorer/context/tags.tsx

```diff
@@ -158,6 +214,41 @@ export const TagsProvider: FC<Prop> = ({children, scope}) => {
       [tagKey]: RemoteDataState.Loading,
     })
 
+    if (isFlagEnabled('v2privateQueryUI') && isIOx) {
+      const queryTextSQL: string = `
+        SELECT DISTINCT("${tagKey}")
+        FROM "${measurement}"
+        ORDER BY "${tagKey}"
+        LIMIT ${DEFAULT_LIMIT}
+      `
+      try {
+        const resp = await queryAPI(queryTextSQL, scope, {
+          language: LanguageType.SQL, // use SQL to get measurement list
+          bucket,
+        } as QueryOptions)
+        const values = (Object.values(resp.parsed.table?.columns ?? [])[0]
+          ?.data ?? []) as string[]
+
```

Comment:
Same comment here re: the type cast.

Reply:
Good point! It makes sense to avoid casting the tag values since the original data type is `string[] | boolean[] | number[]`. Updated in https://github.com/influxdata/ui/pull/6791/commits/689f58c8c69ec61a85ac543f5a8a897cad69f3ee